docker-compose.yml

```diff
@@ -1,133 +1,102 @@
+# This file is designed as a starting point for your production server deployment, not local development work!
 # For a containerized local dev environment, see: https://github.com/mastodon/mastodon/blob/main/README.md#docker
 
-services:
-  db:
-    restart: always
-    image: postgres:14-alpine
-    shm_size: 256mb
-    networks:
-      - internal_network
-    healthcheck:
-      test: ['CMD', 'pg_isready', '-U', 'postgres']
-    volumes:
-      - ./postgres14:/var/lib/postgresql/data
-    environment:
-      - 'POSTGRES_HOST_AUTH_METHOD=trust'
+# You should check the Docker image tag used for the web, sidekiq, and streaming containers to make sure you are running
+# the appropriate version for your deployment. Using `nightly` will deploy the in-development version of Mastodon, which is
+# not considered a stable release. Visit https://github.com/mastodon/mastodon/releases for currently released versions and tags.
+x-image-tag:
+  &image-tag nightly
 
-  redis:
-    restart: always
-    image: redis:7-alpine
-    networks:
-      - internal_network
-    healthcheck:
-      test: ['CMD', 'redis-cli', 'ping']
-    volumes:
-      - ./redis:/data
-
-  # es:
-  #   restart: always
-  #   image: docker.elastic.co/elasticsearch/elasticsearch:7.17.4
-  #   environment:
-  #     - "ES_JAVA_OPTS=-Xms512m -Xmx512m -Des.enforce.bootstrap.checks=true"
-  #     - "xpack.license.self_generated.type=basic"
-  #     - "xpack.security.enabled=false"
-  #     - "xpack.watcher.enabled=false"
-  #     - "xpack.graph.enabled=false"
-  #     - "xpack.ml.enabled=false"
-  #     - "bootstrap.memory_lock=true"
-  #     - "cluster.name=es-mastodon"
-  #     - "discovery.type=single-node"
-  #     - "thread_pool.write.queue_size=1000"
-  #   networks:
-  #      - external_network
-  #      - internal_network
-  #   healthcheck:
-  #      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
-  #   volumes:
-  #      - ./elasticsearch:/usr/share/elasticsearch/data
-  #   ulimits:
-  #     memlock:
-  #       soft: -1
-  #       hard: -1
-  #     nofile:
-  #       soft: 65536
-  #       hard: 65536
-  #   ports:
-  #     - '127.0.0.1:9200:9200'
+  # This is the Mastodon background worker. It processes jobs such fetching posts from other servers and sending e-mails.
+x-sidekiq-base: &sidekiq-base
+  # Remove the comments from this `build` section if you need to build the image from source.
+  # build:
+  #   context: .
+  #   dockerfile: Dockerfile
+  image: ghcr.io/mastodon/mastodon:*image-tag
+  env_file: .env.production
+  environment:
+    - DB_POOL=15
+  restart: always
+  command: bundle exec sidekiq -c 15
+  healthcheck:
+    test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+  networks:
+    - external_network
+    - internal_network
+  # S3 is optional but recommended for storing user-uploaded media files.
+  volumes:
```

Comment:
Could `networks`, `volumes` (maybe `image`, `env_file`… as well) go into a fragment (https://docs.docker.com/reference/compose-file/fragments/) to avoid repeating those?

Something like `&mastodon-ruby-app` or something, for everything common to the Ruby containers